tests/test_cases/test_cocotb/test_handle.py

```diff
@@ -92,36 +88,36 @@ def test_integer(dut):
 
 
 @cocotb.test(expect_error=cocotb.SIM_NAME in ["Icarus Verilog"])
-def test_real_assign_double(dut):
+async def test_real_assign_double(dut):
     """
     Assign a random floating point value, read it back from the DUT and check
     it matches what we assigned
     """
     val = random.uniform(-1e307, 1e307)
     log = logging.getLogger("cocotb.test")
-    yield Timer(1)
+    await Timer(1, "ns")
     log.info("Setting the value %g" % val)
     dut.stream_in_real = val
-    yield Timer(1)
-    yield Timer(1)  # FIXME: Workaround for VHPI scheduling - needs investigation
+    await Timer(1, "ns")
+    await Timer(1, "ns")  # FIXME: Workaround for VHPI scheduling - needs investigation
     got = float(dut.stream_out_real)
     log.info("Read back value %g" % got)
     if got != val:
         raise TestFailure("Values didn't match!")
 
 
 @cocotb.test(expect_error=cocotb.SIM_NAME in ["Icarus Verilog"])
-def test_real_assign_int(dut):
+async def test_real_assign_int(dut):
     """Assign a random integer value to ensure we can write types convertible to
     int, read it back from the DUT and check it matches what we assigned.
     """
     val = random.randint(-2**31, 2**31 - 1)
     log = logging.getLogger("cocotb.test")
-    yield Timer(1)
+    await Timer(1, "ns")
     log.info("Setting the value %i" % val)
     dut.stream_in_real <= val
-    yield Timer(1)
-    yield Timer(1)  # FIXME: Workaround for VHPI scheduling - needs investigation
+    await Timer(1, "ns")
+    await Timer(1, "ns")  # FIXME: Workaround for VHPI scheduling - needs investigation
```

Comment:
I feel like these two are deliberately intended to delay the absolute minimal time, and the comment here is about very short timers.

Reply:
Oh, yes, I'll go over them again.

Reply:
I now also reverted specifying ``units`` in ``test_scheduler/test_kill_coroutine_waiting_on_the_same_trigger`` since it might have been deliberate there as well.